game-core/src/main/java/games/strategy/triplea/ui/menubar/help/UnitStatsTable.java

```diff
@@ -0,0 +1,93 @@
+package games.strategy.triplea.ui.menubar.help;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.ResourceCollection;
+import games.strategy.engine.data.UnitType;
+import games.strategy.triplea.image.UnitImageFactory;
+import games.strategy.triplea.ui.TooltipProperties;
+import games.strategy.triplea.ui.UiContext;
+import games.strategy.triplea.util.TuvUtils;
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class UnitStatsTable {
+
+  public static String getUnitStatsTable(final GameData gameData, final UiContext uiContext) {
+    // html formatted string
+    final StringBuilder hints = new StringBuilder();
+    hints.append("<html>");
+    hints.append("<head><style>th, tr{color:black}</style></head>");
+    try {
+      gameData.acquireReadLock();
+      final Map<GamePlayer, Map<UnitType, ResourceCollection>> costs =
+          TuvUtils.getResourceCostsForTuv(gameData, true);
+      final Map<GamePlayer, List<UnitType>> playerUnitTypes =
+          UnitType.getAllPlayerUnitsWithImages(gameData, uiContext, true);
+      final String color3 = "FEECE2";
+      final String color2 = "BDBDBD";
+      final String color1 = "ABABAB";
+      int i = 0;
+      for (final Map.Entry<GamePlayer, List<UnitType>> entry : playerUnitTypes.entrySet()) {
+        final GamePlayer player = entry.getKey();
+        hints.append("<p><table border=\"1\" bgcolor=\"" + color1 + "\">");
+        hints
+            .append(
+                "<tr><th style=\"font-size:120%;000000\" bgcolor=\"" + color3 + "\" colspan=\"4\">")
+            .append(player == null ? "NULL" : player.getName())
+            .append(" Units</th></tr>");
+        hints
+            .append("<tr")
+            .append(((i & 1) == 0) ? " bgcolor=\"" + color1 + "\"" : " bgcolor=\"" + color2 + "\"")
+            .append("><td>Unit</td><td>Name</td><td>Cost</td><td>Tool Tip</td></tr>");
+        for (final UnitType ut : entry.getValue()) {
+          if (uiContext.getMapData().shouldDrawUnit(ut.getName())) {
+            i++;
+            hints
+                .append("<tr")
+                .append(
+                    ((i & 1) == 0) ? " bgcolor=\"" + color1 + "\"" : " bgcolor=\"" + color2 + "\"")
```

Comment:
Not related to this PR, but `((i & 1) == 0)` is way more complicated than using `i % 2 == 0`

Reply:
Another Side-Note:
The ternary expression could also just be used for the actual color, to avoid the duplicate `bgcolor=` part around it.

Rant Warning!
We use a lot of HTML inside swing, which makes this kind of stuff easier to layout but at the same time I think this is a place where simply going with a lot of swing objects would be the best move.
Also in the HTML world using stuff like bgcolor is only done for legacy browser engines, nowadays you'd just write something like
```
tr {
  background-color: /*color 1 */;
}
tr:nth-child(2n) {
  background-color: /* color 2 */;
}
```

which makes the browser alternate between the colors automatically